snapcraft/internal/build_providers/_base_provider.py

```diff
@@ -259,6 +258,19 @@ def launch_instance(self) -> None:
         # what is on the host
         self._setup_snapcraft()
 
+        # Configure snapd to use http/https proxy.
```

Comment:
How about putting this in it's own method and do some "super" tricks in multipass and lxd to minimize the test spread?